CONTRIBUTING.md

```diff
@@ -133,48 +160,42 @@ A PR is considered to be **ready to merge** when:
   /
   [Maintainers](https://github.com/open-telemetry/community/blob/master/community-membership.md#maintainer)
   or the respective component owner.
-* Major feedbacks are resolved.
+* Major feedback/comments are resolved.
 * It has been open for review for at least one working day. This gives people
   reasonable time to review.
-* Trivial change (typo, cosmetic, doc, etc.) doesn't have to wait for one day.
-* Urgent fix can take exception as long as it has been actively communicated.
-
-Any Maintainer can merge the PR once it is **ready to merge**. Note that some
-PR may not be merged immediately if repo is being in process of a major release
-and the new feature doesn't fit it.
+  * Trivial change (typo, cosmetic, doc, etc.) doesn't have to wait for one day.
+  * Urgent fix can take exception as long as it has been actively communicated.
 
-### How to request for release of package
+Any maintainer can merge PRs once they are **ready to merge** however
+maintainers might decide to wait on merging changes until there are more
+approvals and/or dicussion, or based on other factors such as release timing and
+risk to users. For example if a stable release is planned and a new change is
+introduced adding public API(s) or behavioral changes it might be held until the
+next alpha/beta release.
 
-* Submit a PR with `CHANGELOG.md` file reflecting the version to be released
-along with the date in the following format `yyyy-MMM-dd`.
-
-For example:
-
-```text
-## 1.2.0-beta.2
-
-Released 2022-Jun-21
-```
+## Release process
 
-* Tag the maintainers of this repository
-(@open-telemetry/dotnet-contrib-maintainers) who can release the package.
+For details about the relaese process and information about how to request a
```

Comment:
Can we link to the specific header within releasing.md file, as that doc has lot of maintainers only content too.